server/athenian/api/internal/features/entries.py

```diff
@@ -130,6 +130,38 @@ def _postprocess_cached_facts(
     return result
 
 
+class _PRMetricsLineGHCache:
+    @staticmethod
+    def postprocess(result: np.ndarray, *, metrics: Sequence[str], **kwargs: Any) -> np.ndarray:
+        # cache value has been stored with sorted(metrics), adapt before returning to caller
+        if (sorted_metrics := sorted(metrics)) != metrics:
+            sort_indexes = [metrics.index(m) for m in sorted_metrics]
+            return _sorted_result_metrics(result, sort_indexes)
+
+        return result
+
+    @staticmethod
+    def preprocess(result: np.ndarray, *, metrics: Sequence[str], **kwargs: Any) -> np.ndarray:
+        # cache value must be stored with sorted(metrics), adapt if needed
+        if (sorted_metrics := sorted(metrics)) != metrics:
+            sort_indexes = [sorted_metrics.index(m) for m in metrics]
```

Comment:
```suggestion
            sort_indexes = np.searchsorted(sorted_metrics, metrics)
```
O(n log n) vs. O(n^2)

Reply:
done